.github/workflows/image_build.yml

```diff
@@ -0,0 +1,54 @@
+name: Deploy reference images
+
+on: 
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
```

Comment:
let us only run this pipeline if there are changes in the `function-images` folder. see the linter GH workflow as an example